File: backend/src/main/java/com/example/demo/models/CV.java
package com.example.demo.models;

import jakarta.persistence.*;
import lombok.*;
import java.util.List;

@Entity
@Table(name = "cvs")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor
public class CV {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String profession;
    private Double expectedSalary;
    
    // if info diff from user profiile
    private String contactEmail;
    private String phone;
    private String city;
    private String citizenship;
    private String birthDate;

    @ElementCollection
    @CollectionTable(name = "cv_skills", joinColumns = @JoinColumn(name = "cv_id"))
    @Column(name = "skill")
    private List<String> skills;

    @OneToMany(mappedBy = "cv", cascade = CascadeType.ALL)
    private List<Experience> experiences;

    @OneToMany(mappedBy = "cv", cascade = CascadeType.ALL)
    private List<Education> educations;

    @OneToOne
    @JoinColumn(name = "user_id")
    private User user;
}

File: frontend/src/types/cv.ts
export interface Experience {
  id?: number;
  company: string;
  position: string;
  period?: string | null;
}

export interface Education {
  id?: number;
  institution?: string | null;
  degree?: string | null;
  year?: string | null;
}

export interface Cv {
  id: number;
  profession?: string | null;
  expectedSalary?: number | null;
  contactEmail?: string | null;
  phone?: string | null;
  city?: string | null;
  citizenship?: string | null;
  birthDate?: string | null;
  skills?: string[] | null;
  experiences?: Experience[] | null;
  educations?: Education[] | null;
}

export interface CvSavePayload {
  profession?: string;
  city?: string;
  contactEmail?: string;
  phone?: string;
  skills: string[];
  experiences: Experience[];
  educations: Education[];
}
